scripts/west_commands/runners/nrfjprog.py

```diff
@@ -132,10 +150,21 @@ def do_run(self, command, **kwargs):
                 program_cmd
             ])
         else:
-            if self.family == 'NRF52':
+            if self.family == 'NRF51':
+                commands.append(program_cmd + ['--sectorerase'])
+            elif self.family == 'NRF52':
                 commands.append(program_cmd + ['--sectoranduicrerase'])
             else:
-                commands.append(program_cmd + ['--sectorerase'])
+                is_uicr = {
+                    'NRF53': lambda addr: (0x00FF8000 <= addr < 0x00FF9000) or (0x01FF8000 <= addr < 0x01FF8800),
+                    'NRF91': lambda addr: (0x00FF8000 <= addr < 0x00FF9000),
+                }[self.family]
+
+                if has_region(is_uicr, self.hex_):
+                    # Hex file has UICR contents.
+                    commands.append(program_cmd + ['--chiperase'])
```

Comment:
I assume the build system (CMake) knows that data is being placed in UICR.
If part of the build system decides to place data into UICR, shouldn't it update `runners.yml` section with `--chiperase` information instead ?

Reply:
If you provide a bit of info where UICR is added to hex file, then I can take a look.

Reply:
Search for `provision_PM_HEX_FILE`